### ISSUE TITLE
Edit folder in 1.10 branch fails with exception

### ISSUE THREAD

**Issue by attilah** [OrchardCMS/Orchard#6501]
Repro:
1. create folder
2. select folder
3. edit folder

I think that a missing NOT is the cause of the error in GetRootMediaFolder method:

```
            if (_orchardServices.Authorizer.Authorize(Permissions.ManageMediaContent)) {
                return null;
            }
```

That code reads that if authorized return null.

Exception:
An unhandled exception has occurred and the request was terminated. Please refresh the page. If the error persists, go back
Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object. at Orchard.MediaLibrary.Controllers.FolderController.IsRootFolder(String folderPath) at Orchard.MediaLibrary.Controllers.FolderController.Edit(String folderPath) at lambda_method(Closure , ControllerBase , Object[] ) at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters) at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters) at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters) at System.Web.Mvc.Async.AsyncControllerActionInvoker.ActionInvocation.InvokeSynchronousActionMethod() at System.Web.Mvc.Async.AsyncControllerActionInvoker.<BeginInvokeSynchronousActionMethod>b__39(IAsyncResult asyncResult, ActionInvocation innerInvokeState) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End() at System.Web.Mvc.Async.AsyncResultWrapper.End[TResult](IAsyncResult asyncResult, Object tag) at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3d() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f()

**Comment by attilah** [comment]
As an update on this issue, I think there is a "design" problem, since the GetRootMediaFolder returns null for successful check and in the last return statement too.

IMHO we can treat null as access check failure (I don't know why no exception is thrown), but the root folder should be a default IMediaFolder instance and not a null value.

Also if we are using GetRootMediaFolder for access checks as some comment involve this, then separate the two methods since it is really misleading when reading the code.

**Comment by BenedekFarkas** [comment]
Hey - I've fixed that a few days ago on the 1.9.3 branch (it will be merged into 1.10) in this commit: https://github.com/OrchardCMS/Orchard/commit/b983787d0d9370ec8c1158ee1f66c4214b7f245b

**Comment by BenedekFarkas** [comment]
Fix from 1.9.3 merged into 1.10. Please reopen this, if it's still an issue.
